File: 2-AndroidDesignPatterns/MVP/app/src/main/java/com/example/arch_lab2/model/pojo/Product.java
package com.example.arch_lab2.model.pojo;


import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "products_table")
public class Product {
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }

    public String getThumbnail() {
        return thumbnail;
    }

    public void setThumbnail(String thumbnail) {
        this.thumbnail = thumbnail;
    }

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String title;

    private String description;
    private double price;
    private double rating;
    private String thumbnail;

    // Constructors
    public Product() {
    }

    public Product(String title, String description, double price, double rating, String thumbnail) {
        this.title = title;
        this.description = description;
        this.price = price;
        this.rating = rating;
        this.thumbnail = thumbnail;
    }

}

File: 2-AndroidDesignPatterns/MVVM/app/src/main/java/com/example/mvvm/model/Product.kt
package com.example.mvvm.model

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(tableName = "products")
data class Product(
    @PrimaryKey val id: Int,
    val title: String,
    val description: String,
    val price: Double,
    val rating: Float,
    val category: String,
    val thumbnail: String
) : Parcelable

data class ProductResponse(
    val products: List<Product>
)